### PR TITLE
python312Packages.pygreat: 2019.5.1.dev0 -> 2024.0.2

### PR DESCRIPTION
**Pull request by carlossless**
https://github.com/greatscottgadgets/libgreat/compare/v2019.5.1.dev0...v2024.0.2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/pygreat/default.nix [hunk 1]
```diff
@@ -1,41 +1,52 @@
...
+  '';
+
+  build-system = [
+    setuptools
+  ];
+
+  nativeBuildInputs = [
+    pythonRelaxDepsHook
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/pygreat/default.nix:36]:
```suggestion
```
We don't need explicitly add `pythonRelaxDepsHook` anymore.

File: pkgs/development/python-modules/pygreat/default.nix [hunk 2]
```diff
@@ -1,41 +1,52 @@
...
+  pythonRemoveDeps = [
+    "backports.functools-lru-cache"
+  ];
+
   propagatedBuildInputs = [
```
Comment (natsukium) [inline pkgs/development/python-modules/pygreat/default.nix:41]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/pygreat/default.nix [hunk 3]
```diff
@@ -1,41 +1,52 @@
...
     future
     pyusb
   ];
```
Comment (natsukium) [inline pkgs/development/python-modules/pygreat/default.nix:45]: Could you enable tests with `pytestCheckHook` and `pythonImportsCheck`?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#testing-python-packages-testing-python-packages

**Review by natsukium — DISMISSED** [review]

File: pkgs/development/python-modules/pygreat/default.nix (hunk 3 above)
Comment (carlossless) [reply pkgs/development/python-modules/pygreat/default.nix:40]: I added `pythonImportsCheck`, but this package doesn't have any tests for `pytestCheckHook` currently.

**Comment by carlossless** [comment]
Result of `nixpkgs-review pr 353406` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.greatfet</li>
    <li>python311Packages.greatfet.dist</li>
    <li>python311Packages.pygreat</li>
    <li>python311Packages.pygreat.dist</li>
    <li>python312Packages.greatfet</li>
    <li>python312Packages.greatfet.dist</li>
    <li>python312Packages.pygreat</li>
    <li>python312Packages.pygreat.dist</li>
  </ul>
</details>

**Comment by carlossless** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353406`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.greatfet</li>
    <li>python311Packages.greatfet.dist</li>
    <li>python311Packages.pygreat</li>
    <li>python311Packages.pygreat.dist</li>
    <li>python312Packages.greatfet</li>
    <li>python312Packages.greatfet.dist</li>
    <li>python312Packages.pygreat</li>
    <li>python312Packages.pygreat.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.greatfet</li>
    <li>python311Packages.greatfet.dist</li>
    <li>python311Packages.pygreat</li>
    <li>python311Packages.pygreat.dist</li>
    <li>python312Packages.greatfet</li>
    <li>python312Packages.greatfet.dist</li>
    <li>python312Packages.pygreat</li>
    <li>python312Packages.pygreat.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.greatfet</li>
    <li>python311Packages.greatfet.dist</li>
    <li>python311Packages.pygreat</li>
    <li>python311Packages.pygreat.dist</li>
    <li>python312Packages.greatfet</li>
    <li>python312Packages.greatfet.dist</li>
    <li>python312Packages.pygreat</li>
    <li>python312Packages.pygreat.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.greatfet</li>
    <li>python311Packages.greatfet.dist</li>
    <li>python311Packages.pygreat</li>
    <li>python311Packages.pygreat.dist</li>
    <li>python312Packages.greatfet</li>
    <li>python312Packages.greatfet.dist</li>
    <li>python312Packages.pygreat</li>
    <li>python312Packages.pygreat.dist</li>
  </ul>
</details>

**Comment by melvyn2** [comment]
Result of `nixpkgs-review pr 353406` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.greatfet</li>
    <li>python311Packages.greatfet.dist</li>
    <li>python311Packages.pygreat</li>
    <li>python311Packages.pygreat.dist</li>
    <li>python312Packages.greatfet</li>
    <li>python312Packages.greatfet.dist</li>
    <li>python312Packages.pygreat</li>
    <li>python312Packages.pygreat.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/pygreat/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/pygreat/default.nix:44]: Can we set doCheck explicit to false?

Comment (carlossless) [reply pkgs/development/python-modules/pygreat/default.nix:44]: Done!

**Comment by carlossless** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353406`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.greatfet</li>
    <li>python311Packages.greatfet.dist</li>
    <li>python311Packages.pygreat</li>
    <li>python311Packages.pygreat.dist</li>
    <li>python312Packages.greatfet</li>
    <li>python312Packages.greatfet.dist</li>
    <li>python312Packages.pygreat</li>
    <li>python312Packages.pygreat.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.greatfet</li>
    <li>python311Packages.greatfet.dist</li>
    <li>python311Packages.pygreat</li>
    <li>python311Packages.pygreat.dist</li>
    <li>python312Packages.greatfet</li>
    <li>python312Packages.greatfet.dist</li>
    <li>python312Packages.pygreat</li>
    <li>python312Packages.pygreat.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.greatfet</li>
    <li>python311Packages.greatfet.dist</li>
    <li>python311Packages.pygreat</li>
    <li>python311Packages.pygreat.dist</li>
    <li>python312Packages.greatfet</li>
    <li>python312Packages.greatfet.dist</li>
    <li>python312Packages.pygreat</li>
    <li>python312Packages.pygreat.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 8 packages built:</summary>
  <ul>
    <li>python311Packages.greatfet</li>
    <li>python311Packages.greatfet.dist</li>
    <li>python311Packages.pygreat</li>
    <li>python311Packages.pygreat.dist</li>
    <li>python312Packages.greatfet</li>
    <li>python312Packages.greatfet.dist</li>
    <li>python312Packages.pygreat</li>
    <li>python312Packages.pygreat.dist</li>
  </ul>
</details>

**Review by GaetanLepage — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**